spec/support/macros.rb

```diff
@@ -47,11 +44,23 @@ def with_i18n_fallbacks
       require_fallbacks
 
       around do |example|
-        old_backend = I18n.backend
-        I18n.backend = I18nBackendWithFallbacks.new
+        include I18n::Backend::Fallbacks
+        if I18n.respond_to?(:temp_fallbacks)
+          class << I18n
```

Comment:
This is a little hacky. Turns out. once you include the `I18n::Backend::Fallbacks` module anywhere, the fallbacks methods are defined on I18n, and reassigning the old backend doesn't change that. This is because the `fallbacks` and `fallbacks=` methods are defined outside of the fallbacks module onto I18n itself.

This changed makes it such that the two fallbacks methods are undefined after each test, and redefined later.

Reply:
These methods actually are defined when the backend is loaded (from filesystem), it doesn't even need to be instantiated:

```
serene2% irb -ri18n
irb(main):001:0> I18n.fallbacks
(irb):1:in `<main>': undefined method `fallbacks' for I18n:Module (NoMethodError)
	from /home/w/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /home/w/.rbenv/versions/3.1/bin/irb:25:in `load'
	from /home/w/.rbenv/versions/3.1/bin/irb:25:in `<main>'
irb(main):002:0> require 'i18n/backend/fallbacks'
=> true
irb(main):003:0> I18n.fallbacks
=> {}
```

The presence of `fallbacks` method doesn't mean fallbacks actually are used:
```
irb(main):001:0> I18n.enforce_available_locales = false
=> false
irb(main):002:0> I18n.fallbacks
(irb):2:in `<main>': undefined method `fallbacks' for I18n:Module (NoMethodError)
	from /home/w/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
	from /home/w/.rbenv/versions/3.1/bin/irb:25:in `load'
	from /home/w/.rbenv/versions/3.1/bin/irb:25:in `<main>'
irb(main):003:0> I18n.backend.store_translations('en',{'a'=>'a-en'})
=> {:a=>"a-en"}
irb(main):004:0> I18n.backend.store_translations('ru',{'b'=>'b-ru'})
=> {:b=>"b-ru"}
irb(main):005:0> I18n.t('b')
=> "translation missing: en.b"
irb(main):006:0> require 'i18n/backend/fallbacks'
=> true
irb(main):007:0> I18n.fallbacks
=> {}
irb(main):008:0> I18n.fallbacks[:en]=[:ru]
=> [:ru]
irb(main):009:0> I18n.t('b')
=> "translation missing: en.b"
irb(main):010:0> I18n.backend.class.included_modules.include?(I18n::Backend::Fallbacks)
=> false
irb(main):012:0> I18n.locale='en'
=> "en"
irb(main):013:0> I18n.t('b')
=> "translation missing: en.b"
irb(main):014:0> I18n::Backend::Simple.include(I18n::Backend::Fallbacks)
=> I18n::Backend::Simple
irb(main):015:0> I18n.t('b')
=> "b-ru"
```

The fallbacks logic doesn't identify whether it is active in a sensible way. The best condition for checking whether fallbacks are actually enabled is `I18n.backend.class.included_modules.include?(I18n::Backend::Fallbacks)` as far as I can tell, quite ugly.

What we can do is always test fallbacks in the test configs with the fallbacks enabled, as is the current setup, and add an assertion at the end of a test run that `fallbacks` isn't defined if the test config doesn't have fallbacks set. With this we just need to make sure we don't require any fallback code accidentally and we won't need to go into i18n guts.